Model/Api/CreateOrder.php

```diff
@@ -296,7 +303,10 @@ public function execute(
                 ]]
             ]);
         } finally {
-            $this->response->sendResponse();
+            // we do not want to send a non-error response in the universal case. 
+            if (!$isUniversal) {
+                $this->response->sendResponse();
+            }
```

Comment:
Perhaps each of the exception catch above should be handled differently if $isUniversal == true

Reply:
Potentially yes, that probably makes sense. Going to leave this closed for now though, the refactoring effort was more than anticipated and there were other things with priority. Will come back to this in future.